### DISCUSSION TITLE
More flexibility for the #found# placeholder

### DISCUSSION THREAD

**Discussion by gboye** [2Abendsegler/GClh#2106] [Ideas]
With the emergence of Lab Caches that get logged immediately, it would be convenient to have more flexibility with the #found# placeholder. Allowing for say **#found-5#** or **#found-num#** in general...

At any rate, thanks for gclh, we use it almost every day...

Best,

Gamboy

**Comment by capoaira** [comment]
Thank you for your Idea. 

I see more often logs with excluded Labs more often, so the feature makes sense.
It should be also not a big problem to implement, but I'm not sure, if it makes much sense in that way. Every time you find an ALC you have to change your template. That's IMO more annoying than helpful.
Another possibility that come to my mind is to request the number of found Labs and add the placeholder `#FoundWithoutALC#` and `#FoundALCs`. But this is much traffic, if we have to request it with every log.
@2Abendsegler What do you prefer, or do you have any other ideas?

**Reply by gboye** [reply]
I suppose both implementation would be useful to different people. We count our labs in, so the #FoundWithoutALC# would not work for us but it would for a lot of people. The #Found-Num# would be good for us even if we had to declare several templates to accommodate our evolving situation...

Anyway, thanks for considering the feature,

Regards,

Gamboy

**Comment by 2Abendsegler** [comment]
#FoundWithoutALCs# would then be the same as #Found#, right? However, to make it clearer, we can still include this new parameter.

#FoundALCs# makes sense, good idea. @capoaira, have you ever looked up how to get it? If this is just one request per log to get the number of ALCs found, then that seems fine from my point of view. We could also check whether this parameter is used and only then carry out the request.

If we had another #FoundWithALCs# parameter, then everything would be there, right? The same should apply here as above.

I don't understand exactly what is meant by #found-5# or #found-num#. This could be because we haven't logged any ALCs yet. Isn't that exactly what it means #FoundWithALCs#?

**Reply by gboye** [reply]
What I meant with #found-num# is to get a placeholder based on #found# but with a modifier than gives num less than the original (e.g. If #found# is 120, #found-5# is 115).

This would help people like us who count their ALC in their flow with the period when they are logging caches that came before the ALCs that are already included in #found#. For example, if we are writing the logs of #115 but our #found# is already at 120 due to 1 ALC containing 5 lab caches...  The more ALCs found after the caches we are currently writing for, the more problems with templates...

**Reply by 2Abendsegler** [reply]
I think I understand. It's complicated. So that you don't have to constantly change the templates, you want to set up several templates, for example with the names Signature, Signature-1, Signature-2 ... Signature-5, and with the corresponding found parameters. Depending on how far you are with the logs, you use different templates. Do I understand this in the right manner?

**Reply by gboye** [reply]
Yes. Usually we are say 30 behind for a stretch and then 25 for another, etc. So having different versions of the template like you're saying is exactly what we would like.

**Reply by capoaira** [reply]
> Usually we are say 30 behind for a stretch and then 25 for another, etc. So having different versions of the template like you're saying is exactly what we would like.

This part shouldn't be a problem to implement, but what is if you don't complete an adventure (for example, you found only 3 locations)?

>#FoundWithoutALCs# would then be the same as #Found#, right?

No, #Found# includes all founds (I think we take the information from the header). #FoundWithoutALCs# will be excluding all ALCs (Founds minus ALCs).

> @capoaira, have you ever looked up how to get it?

I hope from the public profile, but I don't test it yet.

**Reply by gboye** [reply]
> > Usually we are say 30 behind for a stretch and then 25 for another, etc. So having different versions of the template like you're saying is exactly what we would like.
> 
> This part shouldn't be a problem to implement, but what is if you don't complete an adventure (for example, you found only 3 locations)?

Then, if we have only these 3 locations (lab caches) in our #found# but are still writing up logs from before, we would use #Found-3# for our current number... (the #Found_no# would be the same as #Found-1#)

What you have in the #Found# placeholder is the number of found caches and  lab locations completed (usually 5 lab caches per ALC), not found caches+ALCs...

**Comment by capoaira** [comment]
I've implemented the requested function for the next release.
You can test the feature here: https://github.com/capoaira/GClh/blob/flexible-found-placeholder/gc_little_helper_II.user.js?raw=true
Here it goes to the current stable release: https://github.com/2Abendsegler/GClh/blob/master/gc_little_helper_II.user.js?raw=true

Is this what you wished for? 

(new ALC placeholder are not included)

**Reply by gboye** [reply]
Yes, it is exactly what we wished for. Thank you so much.
